community/cypher/frontend-3.2/src/main/scala/org/neo4j/cypher/internal/frontend/v3_2/ast/Hint.scala

```diff
@@ -52,11 +52,22 @@ sealed trait LegacyIndexHint extends UsingHint {
   def variables = NonEmptyList(variable)
 }
 
-case class UsingIndexHint(variable: Variable, label: LabelName, properties: Seq[PropertyKeyName])(val position: InputPosition) extends UsingHint with NodeHint {
+sealed trait UsingIndexHintSpec {
```

Comment:
I would prefer the simpler option of just putting `fulfilledByScan: Boolean` on `UsingIndexHint` instead of this Spec model, which seems a little over-engineered to me.

Reply:
The idea was to (_maybe_) make this the default for when you just type `USING INDEX` in 3.4 or 3.5 and then introduce an explicit `USING INDEX SCAN` (if you for some reason need to enforce that behaviour).

Reply:
In any case I expect this to grow beyond and single boolean and it would be nice to have a stable interface, especially since the frontend has moved to a different repository.

Reply:
The frontend has not moved yet, but sure I can buy into that 👍